src/Forms/GridField/GridFieldVersionedState.php

```diff
@@ -0,0 +1,111 @@
+<?php
+namespace SilverStripe\Forms\GridField;
+
+use SilverStripe\Forms\GridField\GridField;
+use SilverStripe\Forms\GridField\GridField_ColumnProvider;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Versioned\Versioned;
+use SilverStripe\Core\Convert;
+
+class GridFieldVersionedState implements GridField_ColumnProvider
+{
+    protected $column = null;
+
+    protected $versionedLabelFields = ['Name', 'Title'];
```

Comment:
these could be set in a constructor and updated with setter/getters so it's more versatile for changes